airgun/entities/config_report.py

```diff
@@ -0,0 +1,68 @@
+from airgun.entities.base import BaseEntity
+from airgun.navigation import NavigateStep
+from airgun.navigation import navigator
+from airgun.utils import retry_navigation
+from airgun.views.config_report import ConfigReportDetailsView
+from airgun.views.config_report import ConfigReportsView
+
+
+class ConfigReportEntity(BaseEntity):
+    endpoint_path = '/config_reports'
+
+    def read(self, widget_names=None, host_name=None):
+        """Read all values for generated Config Reports"""
+        view = self.navigate_to(self, 'All')
+        view.search(host_name)
+        return view.read(widget_names=widget_names)
+
+    def search(self, hostname):
+        """Search for specific Config report"""
+        view = self.navigate_to(self, 'Report Details', host_name=hostname)
+        return view.read()
+
+    def export(self, host_name=None):
+        """Export a Config report.
+
+        :return str: path to saved file
+        """
+        view = self.navigate_to(self, 'All')
+        view.search(host_name)
```

Comment:
you can add check if the hostname exists then you wanted to search otherwise no need to search here. 
`python 
if host_name:
  view.search(host_name)
`